src/js/angular/core/services/license.service.js

```diff
@@ -1,47 +1,198 @@
-        } else if (that.productType === "graphdb") {
-            that.productTypeHuman = "GraphDB";
+    };
+
+    const updateProductType = (license) => {
+        _productType = license.productType;
+        if (_productType === "standard") {
+            _productTypeHuman = "Standard";
+        } else if (_productType === "enterprise") {
+            _productTypeHuman = "Enterprise";
+        } else if (_productType === "free") {
+            _productTypeHuman = "Free";
+        } else if (_productType === "graphdb") {
+            _productTypeHuman = "GraphDB";
         }
     };
 
-    this.checkLicenseStatus();
+    /**
+     * Updates the tracking behavior based on the user's tracking consent or permission.
+     * @private
+     */
+    const updateTracking = () => {
+        if (isTrackingAllowed()) {
+            addGoogleTagManagerScript();
+        } else {
+            removeGoogleTagManagerScripts();
+        }
+    };
+
+    /**
+     * Determines if tracking is allowed based on license status and product type.
+     * @return {boolean} A promise that resolves to a boolean indicating if tracking is allowed.
+     */
+    const isTrackingAllowed = () => {
+        const licenseTypeOfUse = _license && _license.typeOfUse.toLowerCase();
+        return _productType === "free" ||
+            licenseTypeOfUse === EVALUATION_TYPE_1 ||
+            licenseTypeOfUse === EVALUATION_TYPE_2;
+    };
+
+    /**
+     * Adds the Google Tag Manager (GTM) script to the document's head if it is not already present.
+     * This function creates a new script element for GTM and appends it to the document's head.
+     * @private
+     */
+    const addGoogleTagManagerScript = () => {
+        if (!gtmScriptExists()) {
+            const dataLayerScript = $document[0].createElement('script');
+            dataLayerScript.text = "(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start': new Date().getTime(),event:'gtm.js'}); var f=d.getElementsByTagName(s)[0], j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:''; j.async=true; j.src='https://www.googletagmanager.com/gtm.js?id='+i+dl; f.parentNode.insertBefore(j,f); })(window,document,'script','dataLayer','GTM-WBP6C6Z4');";
+            $document[0].getElementsByTagName('head')[0].appendChild(dataLayerScript);
+        }
+    };
+
+    /**
+     * Checks if a Google Tag Manager (GTM) script is already present in the document.
+     *
+     * This function searches through all `<script>` elements in the document and returns `true` if any script
+     * has a `src` attribute that contains 'googletagmanager'. It returns `false` otherwise.
+     *
+     * @return {boolean} - Returns `true` if a GTM script is found, otherwise `false`.
+     * @private
+     */
+    const gtmScriptExists = () => {
+        const scripts = $document[0].getElementsByTagName('script');
+        return Array.from(scripts).some((script) => script.src && script.src.includes('googletagmanager'));
```

Comment:
Two things here:
* You add the gtm script in the head, but here you are searching for it in the entire document which a bit resource waisting because all app scripts are appended in the body and they are huge. So, you can get the scripts only from the head tag where there just a few of them and they are small relative to bundle scripts.
* Why don't you add an attribute (id for example) to the gtm script tag and just select it instead of searching in scripts content as you do currently.

Reply:
When the GTM script is injected, it dynamically adds additional scripts depending on how many tags
are configured within your GTM container. These scripts are usually inserted into the `<head>` section,
but it’s better to search through the entire document to ensure all related GTM scripts are found.
Added to js docs also